pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -185,20 +186,16 @@ def _run_optimization(
         else:
             if self.trainer.lightning_module.automatic_optimization:
                 self._optimizer_step(optimizer, opt_idx, batch_idx, closure)
-                if len(self.trainer.optimizers) > 1:
-                    # revert back to previous state
-                    self.trainer.lightning_module.untoggle_optimizer(opt_idx)
             else:
                 result = self._training_step(split_batch, batch_idx, opt_idx, self._hiddens)
 
-            if not result:
-                # user decided to skip optimization
-                return result
-
+        if result:
             # update running loss + reset accumulated loss
             self._update_running_loss(result.loss)
+            self._process_closure_result(result)
 
-        self._process_closure_result(result)
+        # untoggle model params
+        self._run_optimization_end(opt_idx)
```

Comment:
reviewer: untoggling happens here.

on master, this happens only in the "else" branch when not accumulating.